OpenRA.Mods.Common/Traits/Crates/GrantExternalConditionCrateAction.cs

```diff
@@ -74,7 +77,8 @@ public override void Activate(Actor collector)
 						.FirstOrDefault(t => t.Info.Condition == info.Condition && t.CanGrantCondition(a, self));
 
 					if (external != null)
-						external.GrantCondition(a, self, info.Duration);
+						for (var n = 0; n < info.Levels; n++)
+							external.GrantCondition(a, self, info.Duration);
```

Comment:
Granting multiple times won't necessarily work properly if `ExternalCondition` defines level caps.  We need to be checking `CanGrantCondition` inside the loop!